src/arr/compiler/type-check-structs.arr

```diff
@@ -431,8 +431,8 @@ sharing:
   method solve-level-helper(self, solution :: ConstraintSolution, context :: Context) -> FoldResult<{ConstraintSystem; ConstraintSolution}>:
     solve-helper-constraints(self, solution, context).bind(lam({system; shadow solution}, shadow context):
       solve-helper-refinements(system, solution, context).bind(lam({shadow system; shadow solution}, shadow context):
-        solve-helper-fields(system, solution, context).bind(lam({shadow system; shadow solution}, shadow context):
-          solve-helper-examples(system, solution, context)
+        solve-helper-examples(system, solution, context).bind(lam({shadow system; shadow solution}, shadow context):
```

Comment:
Why the swap in order, here?  Is there a semantic change as a result?

Reply:
`solve-helper-fields` checks to make sure that all types required to have certain fields actually have those fields. So it has to come after all the types are solved for. Therefore it should come after `solve-helper-examples`. I think with how they actually are used it makes no difference to the order, but this is the more correct order.